foldamers/parameters/angle_distributions.py

```diff
@@ -0,0 +1,308 @@
+    :param pdbfile: path to pdb trajectory file
+    :type pdbfile: str
+    
+    :param nbins: number of bins spanning the range of 0 to 180 degrees, default = 90
+    :type nbins: int
+    
+    :param plotfile: Base filename for saving bond angle distribution pdf plots
+    :type plotfile: str
+    
+    """
+    
+    # Load in a trajectory pdb file:
+    traj = md.load(pdbfile)
+    nframes = traj.n_frames
+    
+    # Get angle list
+    angle_list = CGModel.get_bond_angle_list(cgmodel)
+    ang_types = [] # List of angle types for each angle in angle_list
+    ang_array = np.zeros((len(angle_list),3))
+    
+    # Angle types:
+    # 1 bb-bb-bb
+    # 2 bb-bb-sc (sc-bb-bb)
+    # 3 bb-sc-sc (sc-sc-bb)
+    # 4 sc-sc-sc
+    
+    # Assign angle types:
+    
+    for i in range(len(angle_list)):
+        ang_array[i,0] = angle_list[i][0]
+        ang_array[i,1] = angle_list[i][1]
+        ang_array[i,2] = angle_list[i][2]
+        at1_type = CGModel.get_particle_type(cgmodel,angle_list[i][0])
+        at2_type = CGModel.get_particle_type(cgmodel,angle_list[i][1])
+        at3_type = CGModel.get_particle_type(cgmodel,angle_list[i][2])
+        
+        if at1_type == 'backbone': # bb-X-X
+            if at2_type == 'backbone': # bb-bb-X
+                if at3_type == 'backbone': # bb-bb-bb
+                    ang_types.append(1)
+                    
+                elif at3_type == 'sidechain': # bb-bb-sc
+                    ang_types.append(2)
+                    
+            elif at2_type == 'sidechain': # bb-sc-X
+                ang_types.append(3) # bb-sc-sc
+                
+        elif at1_type == 'sidechain': # sc-X-X
+            if at2_type == 'backbone': # sc-bb-X
+                ang_types.append(2) # sc-bb-bb
+                
+            elif at2_type == 'sidechain': # sc-sc-X
+                if at3_type == 'backbone': # sc-sc-bb
+                    ang_types.append(3)
+                    
+                elif at3_type == 'sidechain': # sc-sc-sc
+                    ang_types.append(4)
+                    
+    # Sort angles by type into separate arrays for mdtraj compute_angles
+    # There's probably a more efficient way to do this
+    ang1_array = np.zeros((ang_types.count(1),3))
+    ang2_array = np.zeros((ang_types.count(2),3))
+    ang3_array = np.zeros((ang_types.count(3),3))
+    ang4_array = np.zeros((ang_types.count(4),3))
+
+    n1 = 0
+    n2 = 0
+    n3 = 0
+    n4 = 0
+    
+    for i in range(len(angle_list)):
+        if ang_types[i] == 1:
+            ang1_array[n1,:] = ang_array[i,:]
+            n1 += 1
+        if ang_types[i] == 2:
+            ang2_array[n2,:] = ang_array[i,:]
+            n2 += 1
+        if ang_types[i] == 3:
+            ang3_array[n3,:] = ang_array[i,:]
+            n3 += 1
+        if ang_types[i] == 4:
+            ang4_array[n4,:] = ang_array[i,:]    
+            n4 += 1
+    
+    # Save sub_arrays for each angle type to dictionary
+    ang_sub_arrays = {}

```

Comment:
Think about ways to loop over an list.  For example, a list of strings ["bb_bb_bb","bb_bb_sc","bb_sc_sc"], and arrays n and ang_array, syo 

for array in ang_sub_arrays:
            if n[array] > 0:
                    ang_sub_array[array] = ang_array.